emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/AttributeValueSerializer.java

```diff
@@ -0,0 +1,97 @@
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+
+public class AttributeValueSerializer implements JsonSerializer<AttributeValue> {
+
+  @Override
+  public JsonElement serialize(AttributeValue attributeValue, Type type,
+                               JsonSerializationContext context) {
+    if (attributeValue == null) {
+      return JsonNull.INSTANCE;
+    }
+
+    JsonObject serializedValue = new JsonObject();
+    switch (attributeValue.type()) {
+      case B:
+        serializedValue.add(toV1FieldCasingStyle(DynamoDBTypeConstants.BINARY),
+            context.serialize(attributeValue.b()));
+        break;
+      case BOOL:
+        serializedValue.add(toV1FieldCasingStyle(DynamoDBTypeConstants.BOOLEAN),
+            new JsonPrimitive(attributeValue.bool()));
+        break;
+      case BS:
+        JsonArray sdkBytesList = new JsonArray();
+        attributeValue.bs()
+            .forEach(item -> sdkBytesList.add(context.serialize(item, SdkBytes.class)));
+        serializedValue.add(toV1FieldCasingStyle(DynamoDBTypeConstants.BINARY_SET), sdkBytesList);
+        break;
+      case L:
+        JsonArray attributeList = new JsonArray();
+        attributeValue.l()
+            .forEach(item -> attributeList.add(context.serialize(item, AttributeValue.class)));
+        serializedValue.add(toV1FieldCasingStyle(DynamoDBTypeConstants.LIST), attributeList);
+        break;
+      case M:
+        JsonObject avm = new JsonObject();
+        attributeValue.m().entrySet()
+            .forEach(entry ->
+                avm.add(entry.getKey(), context.serialize(entry.getValue(), AttributeValue.class)));
+        serializedValue.add(toV1FieldCasingStyle(DynamoDBTypeConstants.MAP), avm);
+        break;
+      case N:
+        serializedValue.add(toV1FieldCasingStyle(DynamoDBTypeConstants.NUMBER),
+            new JsonPrimitive(attributeValue.n()));
+        break;
+      case NS:
+        JsonArray numberList = new JsonArray();
+        attributeValue.ns().forEach(item -> numberList.add(new JsonPrimitive(item)));
+        serializedValue.add(toV1FieldCasingStyle(DynamoDBTypeConstants.NUMBER_SET), numberList);
+        break;
+      case NUL:
+        serializedValue.add(toV1FieldCasingStyle(DynamoDBTypeConstants.NULL),
+            new JsonPrimitive(attributeValue.nul()));
+        break;
+      case S:
+        serializedValue.add(toV1FieldCasingStyle(DynamoDBTypeConstants.STRING),
+            new JsonPrimitive(attributeValue.s()));
+        break;
+      case SS:
+        JsonArray stringList = new JsonArray();
+        attributeValue.ss().forEach(item -> stringList.add(new JsonPrimitive(item)));
+        serializedValue.add(toV1FieldCasingStyle(DynamoDBTypeConstants.STRING_SET), stringList);
+        break;
+      default:
+        break;
+    }
+    return serializedValue;
+  }
+
+  private static String toV1FieldCasingStyle(String typeConstant) {
```

Comment:
Not sure I'm imaging the correct scenario: the customer data provided by other components such as Hadoop and Hive will be deserialized to V1 style so that it's compatible with  customers' previous data. Is it correct?

Reply:
Yes, it is for serialization here, to keep backward compatibility with the previous version.